### ISSUE TITLE
Only show search result tabs by default

### ISSUE THREAD

**Issue by mYnDstrEAm** [NiklasGollenstede/tst-search#28]
Thanks for your addon, it's very useful - I learned about it at https://github.com/piroor/treestyletab/issues/3332

By default it still shows other tabs instead of only the search results. One can hide all tabs not in the search results in the addon settings by setting **Result Highlighting->Other/Misses->Hide Completely**. Could this become either the default or could the user get prompted about which setting setting is preferred after installation / at first usage?

Other users also asked about this [in the addon reviews](https://addons.mozilla.org/en-US/firefox/addon/tst-search/reviews/) (where I learned about this option).

I think most users are more likely to use this by default. If you have a lot of tabs even the very small tab heights make the results view badly navigatable. It may also be better for performance, but I haven't checked resource use. It would be a simple change to make this other option the default.

----

Furthermore:

* When changing this setting it had some delay, I think it should lock changing the setting for a 10 seconds or s and immediately change the boxes checkmark because otherwise it may get clicked multiple times in a row or get changed after the preferences page is left.

* A main use-case of this addon is displaying all tabs of a specified domain, not searching per se. When the user has this use-case in mind the addon may not be so easy to find. I guess one could at least add something like "display all tabs of a specific domain/website" to the readme near to where it says "Search by full text" and/or to the addon description at Mozilla.

* This may be a separate issue and probably isn't important if it can't be implemented quickly: it would be nice if the addon could show _the search results only_ in the separate subpanel so you can see the tabs in their original context in the main tabs panel and the search results tabs on the subpanel. (When you click on a search result tab it's visible in two panels at once. Maybe still having contextual tabs visible was the original motivation for why this showing only search result tabs isn't the default.)

**Comment by gabriel-andreescu** [comment]
When I first discovered this addon, I instantly removed it when I saw how it's behaving when searching, recently I got annoyed by searching in the address bar for tabs, again, and thought I could maybe search for this addons source code to change it myself, this is when I found out it actually has a setting to disable the weird shrinking it does by default.

TLDR: I highly suggest switching the 'Hide Completely' on by default, it's a UX you see a lot more often then weirdly squiggled irrelevant search results.

Also, thank you for the work you put into this so we don't have to!

**Comment by mastoca** [comment]
I too turned Hide Completely on as soon as I found it. At first I thought the extension was broken or had css issues. I didn't uninstall right away as I had seen it being used in a stream and knew it was better than that.
